### PR TITLE
Fix Seq.drop.

### PR DESCRIPTION
**Pull request by Choc13**
Addresses #468

### CODE REVIEW

File: src/FSharpPlus/Extensions/Seq.fs [hunk 1]
```diff
@@ -168,10 +168,12 @@ module Seq =
     ///
     /// <returns>The result sequence.</returns>
     let drop count (source: seq<_>) =
-        let mutable i = count
-        use e = source.GetEnumerator ()
-        while (i > 0 && e.MoveNext ()) do i <- i-1
-        seq { while e.MoveNext () do yield e.Current }
+        seq {
```
Comment (Choc13) [inline src/FSharpPlus/Extensions/Seq.fs:171]: If this implementation is less efficient/desirable for some reason (although I don't believe it's creating any additional overhead and it's still lazy) then another solution that is closer to the old code is

```fsharp
let drop count (source: seq<_>) =
    seq {
        let mutable i = count
        use e = source.GetEnumerator ()
        while (i > 0 && e.MoveNext ()) do i <- i-1
        while e.MoveNext () do yield e.Current 
    }
```

As by moving the `seq` around the `use` statement on `GetEnumerator()` it means that it won't be disposed until the `seq` computation has completed.

Comment (gusty) [reply src/FSharpPlus/Extensions/Seq.fs:171]: Yes, I think that was the original intention in the existing code.

**Comment by gusty** [comment]
Thanks for spotting and fixing this.

**Comment by Choc13** [comment]
No problem, thanks for merging so fast. Any ideas when it will be in a release?

**Comment by gusty** [comment]
We're trying to figure out if we can fix something else and then we'll release

**Pull request merged**